dashboards-observability/public/components/application_analytics/components/app_table.tsx

```diff
@@ -0,0 +1,246 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import {
+    EuiButton,
+    EuiContextMenuItem,
+    EuiContextMenuPanel,
+    EuiFlexGroup,
+    EuiFlexItem,
+    EuiHorizontalRule,
+    EuiInMemoryTable,
+    EuiLink,
+    EuiPage,
+    EuiPageBody,
+    EuiPageContent,
+    EuiPageContentHeader,
+    EuiPageContentHeaderSection,
+    EuiPageHeader,
+    EuiPageHeaderSection,
+    EuiPopover,
+    EuiSpacer,
+    EuiTableFieldDataColumnType,
+    EuiText,
+    EuiTitle,
+  } from '@elastic/eui';
+import _ from 'lodash';
+import { TraceAnalyticsComponentDeps } from 'public/components/trace_analytics/home';
+import React, { ReactElement, useEffect, useState } from 'react';
+import { ApplicationType } from '../home';
+
+interface AppTableProps extends TraceAnalyticsComponentDeps {
+    loading: boolean;
+    applications: Array<ApplicationType>;
+  };
+
+export function AppTable(props: AppTableProps) {
+  const [isActionsPopoverOpen, setIsActionsPopoverOpen] = useState(false);
+  const { applications, parentBreadcrumb } = props;
+
+  useEffect(() => {
+    props.chrome.setBreadcrumbs(
+      [
+      parentBreadcrumb,
+      {
+        text: 'Application analytics',
+        href: '#/application_analytics',
+      }
+    ]);
+  })
+
+  const popoverButton = (
+    <EuiButton
+      iconType="arrowDown"
+      iconSide="right"
+      onClick={() => setIsActionsPopoverOpen(!isActionsPopoverOpen)}
+    >
+      Actions
+    </EuiButton>
+  );
+
+  const popoverItems: ReactElement[] = [
+    <EuiContextMenuItem
+      key="rename"
+      disabled={applications.length === 0}
+    >
+      Rename
+    </EuiContextMenuItem>,
+    <EuiContextMenuItem
+      key="duplicate"
+      disabled={applications.length === 0}
+    >
+      Duplicate
+    </EuiContextMenuItem>,
+    <EuiContextMenuItem
+      key="delete"
+      disabled={applications.length === 0}
+    >
+      Delete
+    </EuiContextMenuItem>,
+    <EuiContextMenuItem
+      key="addSample"
+      disabled={applications.length === 0}
+    >
+      Add sample application
+    </EuiContextMenuItem>,
+  ];
+
+  const tableColumns = [
+    {
+      field: 'name',
+      name: 'Name',
+      sortable: true,
+      truncateText: true,
+      render: (value, record) => (
+        // <EuiLink href={`#/application_analytics/${record.id}`}>{_.truncate(value, { length: 100 })}</EuiLink>
+        <EuiLink href={`#/application_analytics/id`}>{_.truncate(value, { length: 100 })}</EuiLink>
```

Comment:
`id` is hard coded?

Reply:
For when I didn't have any data to work with it was, this will change once I connect the backend.